modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/index.d.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+export function FrontendDataSet({
+	actionParameterName,
+	activeViewSettings,
+	apiURL,
+	appURL,
+	bulkActions,
+	creationMenu,
+	currentURL,
+	customDataRenderers,
+	customViewsEnabled,
+	filters,
+	formId,
+	formName,
+	id,
+	initialSelectedItemsValues,
+	inlineAddingSettings,
+	inlineEditingSettings,
+	items,
+	itemsActions,
+	namespace,
+	nestedItemsKey,
+	nestedItemsReferenceKey,
+	onActionDropdownItemClick,
+	onBulkActionItemClick,
+	overrideEmptyResultView,
+	pagination,
+	portletId,
+	selectedItemsKey,
+	selectionType,
+	showManagementBar,
+	showPagination,
+	showSearch,
+	sidePanelId,
+	sorting,
+	style,
+	views,
+}: IFrontendDataSetProps): JSX.Element;
+
+interface IFrontendDataSetProps {
+	actionParameterName?: string;
+	activeViewSettings?: string;
+	apiURL?: string;
+	appURL?: string;
+	bulkActions?: any[];
+	creationMenu?: {
+		primaryItems?: any[];
+		secondaryItems?: any[];
+	};
+	currentURL?: string;
+	customDataRenderers?: any;
+	customViewsEnabled?: boolean;
+	enableInlineAddModeSetting?: {
+		defaultBodyContent?: object;
+	};
+	filters?: any[];
+	formId?: string;
+	formName?: string;
+	id: string;
+	initialSelectedItemsValues?: any[];
+	inlineAddingSettings?: {
+		apiURL: string;
+		defaultBodyContent: object;
+	};
+	inlineEditingSettings?: [
+		boolean,
+		{
+			alwaysOn: boolean;
+			defaultBodyContent: object;
+		}
+	];
+	items?: any[];
+	itemsActions?: any[];
```

Comment:
Hey @danvb112, I think you can improve those types on itemsActions by taking a look on API table at https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/2012217428/Frontend+Data+Set+Actions#Actions-Configuration